src/openmrs-fetch.test.ts

```diff
@@ -175,6 +184,34 @@ describe("openmrsFetch", () => {
         expect(err.response.status).toBe(400);
       });
   });
+
+  it(`redirects to login page when the server responds with a 401`, done => {
+    // @ts-ignore
+    window.location.assign = jest.fn();
+    mockGetConfig.mockReturnValueOnce({
+      redirectAuthFailure: { enabled: true, url: "test/url", errors: [401] }
+    });
+
+    // @ts-ignore
+    window.fetch.mockReturnValue(
+      Promise.resolve({
+        ok: false,
+        status: 401,
+        statusText: "You are not authorized",
+        text: () => Promise.resolve("a string response body")
+      })
+    );
+
+    return openmrsFetch("/ws/rest/v1/session")
+      .then(data => {
+        fail("Promise shouldn't resolve when server responds with 401");
+      })
+      .catch(err => {
+        // @ts-ignore
+        expect(window.location.assign.mock.calls[0][0]).toBe("test/url");
+        done();
```

Comment:
When you return a promise from a jest function, you don't need to call `done()` when you're done. It knows just to wait for the promise to fulfill.

See https://jestjs.io/docs/en/asynchronous#promises

```suggestion
```